omod/src/main/java/org/openmrs/module/registrationapp/page/controller/RegisterPatientPageController.java

```diff
@@ -36,6 +33,15 @@ public void get(UiSessionContext sessionContext, PageModel model,
     }
 
     public void addModelAttributes(PageModel model, Patient patient, AppDescriptor app, PatientIdentifierType primaryIdentifierType, String breadcrumbOverride) throws Exception {
+        // check if the appId needs to be replaced with the custom one defined as a global property
+        if (RegistrationAppUiUtils.getDefaultRegistrationAppId().equals(app.getId())) {
+            // replace the app with the one configured for registration
+            AppDescriptor customApp = Context.getService(AppFrameworkService.class).getApp(RegistrationAppUiUtils.getDefaultRegistrationAppId());
+            if (customApp != null) {
```

Comment:
Can this ever be null? I thought if there is nothing configured, I thought the method call above returns the default.